apps/dcos_net/src/dcos_net_mesos_listener.erl

```diff
@@ -986,14 +1001,106 @@ stream(Data, #state{pid=Pid, size=Size, buf=Buf}=State) ->
     Buf0 = <<Buf/binary, Data/binary>>,
     case byte_size(Buf0) of
         BufSize when BufSize >= Size ->
-            <<Head:Size/binary, Tail/binary>> = Buf0,
-            State0 = State#state{size=undefined, buf=Tail},
-            try jiffy:decode(Head, [return_maps]) of Obj ->
-                {next, Obj, State0}
-            catch error:Error ->
-                {error, Error}
-            end;
+            stream_decode(Buf0, Size, State);
         _BufSize ->
             httpc:stream_next(Pid),
             {next, State#state{buf=Buf0}}
     end.
+
+-spec(stream_decode(binary(), integer(), State) -> {error, term()} |
+     {next, State} | {next, jiffy:object(), State}
+         when State :: state()).
+stream_decode(Buf, Size, State) ->
+    <<Head:Size/binary, Tail/binary>> = Buf,
+    State0 = State#state{size=undefined, buf=Tail},
+    try
+        prometheus_counter:inc(mesos_listener, messages_total, [], 1),
+        prometheus_counter:inc(mesos_listener, bytes_total, [], Size)
+    catch error:_Error ->
+        ok
+    end,
+    try jiffy:decode(Head, [return_maps]) of Obj ->
+        {next, Obj, State0}
+    catch error:Error ->
+        {error, Error}
+    end.
+
+
+%%%===================================================================
+%%% Metrics functions
+%%%===================================================================
+
+-spec(init_metrics() -> ok).
+init_metrics() ->
+    prometheus_summary:new([
```

Comment:
@urbanserj what would trigger a subscriber to be added here? I have a minidcos cluster with one agent and one master. I have 0 subscribers. My cluster is running some apps, including one app with a VIP that keeps restarting so I could keep receiving status updates. But the `map:size(Subs)` is 0. 

I am looking to test if that metric is being reported properly.

Reply:
You should have 1 subscriber on `leader.mesos`, it's `dcos_dns_mesos` [1].

[1] https://github.com/dcos/dcos-net/blob/master/apps/dcos_dns/src/dcos_dns_mesos.erl#L76